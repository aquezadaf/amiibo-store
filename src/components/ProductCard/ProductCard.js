import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next';
import styles from './ProductCard.module.css'

function ProductCard(props) {
    const { image, name, type, price, onPress } = props
    const { t } = useTranslation();
    return (
        <div className={styles.container}>
            <img className={styles.image} src={image} alt="amiibo" />
            <div className={styles.product}>
                <div className={styles.name}>{name}</div>
                <div className={styles.type}>{type}</div>
                <div className={styles.price}>$ {price}</div>
                <button className={styles.button} onClick={() => onPress(props)}>{t('product.addToCart')}</button>
            </div>
        </div>
    );
}

ProductCard.propTypes = {
    id: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    onPress: PropTypes.func.isRequired
}

export default ProductCard