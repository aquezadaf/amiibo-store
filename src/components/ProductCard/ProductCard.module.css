.container {
    width: 250px;
    height: 350px;
    border-radius: 10px;
    background-color: #fbf9f9;
    margin: 10px;
}

.container:hover {
    box-shadow: 0px 5px 31px -1px rgba(0, 0, 0, 0.15);
    transition: 200ms ease-out;
}

.picture {
    height: 200px;
    margin: 10px;
}

.product {
    background-color: #fffdff;
    height: 110px;
    padding: 10px;
}

.name {
    font-size: 15px;
    color: #393c45;
    font-weight: 700;
    text-align: left;
}

.type {
    font-size: 15px;
    color: #b3b3b3;
    padding: 2px 0 5px 0;
    text-align: left;
}

.price {
    color: #393c45;
    font-size: 20px;
    font-weight: 600;
    text-align: left;
    padding-bottom: 8px;
}

.button {
    width: 152px;
    font-size: 15px;
    border: 2px solid #fff;
    background-color: #00b7ff;
    text-align: center;
    text-transform: uppercase;
    font-weight: 700;
    padding: 8px 0;
}
