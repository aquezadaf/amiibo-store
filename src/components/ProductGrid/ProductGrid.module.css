.container {
    margin: 10px;
}

.title {
    margin-left: 10px;
    font-size: 30px;
    font-weight: bold;
    text-align: left;
}

.products {
    display: flex;
    flex-wrap: wrap;
}
