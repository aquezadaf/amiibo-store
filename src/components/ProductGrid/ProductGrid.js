import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next';
import styles from './ProductGrid.module.css'
import ProductCard from '../ProductCard'

function ProductGrid(props) {
    const { products } = props
    const { t } = useTranslation()
    return (
        <div className={styles.container}>
            <div className={styles.title}>{t('productGrid.title')}</div>
            <div className={styles.products}>
                {products.map(product => <ProductCard key={product.id} {...product} />)}
            </div>
        </div>
    )
}

ProductGrid.propTypes = {
    products: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired
    }))
}

export default ProductGrid
