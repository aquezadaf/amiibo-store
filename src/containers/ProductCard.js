import { connect } from 'react-redux'
import ProductCard from '../components/ProductCard'
import actions from '../actions'

const mapDispatchToProps = dispatch => ({
    onPress: (product) =>  dispatch(actions.shoppingCart.addProductToCart(product))
});

export default connect(null, mapDispatchToProps)(ProductCard);
