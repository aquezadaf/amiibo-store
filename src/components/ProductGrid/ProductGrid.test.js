import React from 'react'
import { shallow } from 'enzyme'
import ProductGrid from './ProductGrid'
import ProductCard from "../../containers/ProductCard";

jest.mock("../../containers/ProductCard", () => () => null);

const defaultProps = {
    products: [{
        id: '01',
        image: 'http://...',
        name: 'Name',
        type: 'Type',
        price: 1000
  }]
}

it('renders product grid', () => {
    const wrapper = shallow(<ProductGrid {...defaultProps} />)
    expect(wrapper.html()).toMatchSnapshot()
})

it('passes products to product card', () => {
    const wrapper = shallow(<ProductGrid {...defaultProps} />)
    const props = wrapper.find(ProductCard).at(0).props()
    expect(props).toEqual(defaultProps.products[0])
})
