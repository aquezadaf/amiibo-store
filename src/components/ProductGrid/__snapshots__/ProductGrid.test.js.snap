// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product grid 1`] = `"<div class=\\"container\\"><div class=\\"title\\">productGrid.title</div><div class=\\"products\\"></div></div>"`;
